Enhance documentation for `LocalHistoryEntry` class

AbdeMohlbi opened the pull request:
Fixes : #11660

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

**Note**: The Flutter team is currently trialing the use of [Gemini Code Assist for GitHub](https://developers.google.com/gemini-code-assist/docs/review-github-code). Comments from the `gemini-code-assist` bot should not be taken as authoritative feedback from the Flutter team. If you find its comments useful you can update your code accordingly, but if you are unsure or disagree with the feedback, please feel free to wait for a Flutter team member's review for guidance on which automated comments should be addressed.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

AbdeMohlbi committed:
Enhance documentation for LocalHistoryEntry class

Added documentation for LocalHistoryEntry class, including usage examples and references to related classes and methods.

AbdeMohlbi committed:
the analyzer has a valid point

AbdeMohlbi committed:
Update routes.dart

AbdeMohlbi committed:
Fix formatting of 'See also' section in routes.dart

victorsanni commented on packages/flutter/lib/src/widgets/routes.dart:
````diff
@@ -679,6 +679,98 @@ abstract interface class PredictiveBackRoute {
+/// ```
+/// {@end-tool}
+///
+/// See also
````
```suggestion
/// See also:
```

AbdeMohlbi committed:
Update packages/flutter/lib/src/widgets/routes.dart

Co-authored-by: Victor Sanni <victorsanniay@gmail.com>

AbdeMohlbi committed:
Update packages/flutter/lib/src/widgets/routes.dart

Co-authored-by: Victor Sanni <victorsanniay@gmail.com>

AbdeMohlbi committed:
Update packages/flutter/lib/src/widgets/routes.dart

Co-authored-by: Victor Sanni <victorsanniay@gmail.com>

victorsanni committed:
Merge branch 'master' into abdemohlbi_issue_11660

AbdeMohlbi committed:
Merge branch 'master' into abdemohlbi_issue_11660

victorsanni commented on packages/flutter/lib/src/widgets/routes.dart:
````diff
@@ -679,6 +679,96 @@ abstract interface class PredictiveBackRoute {
+/// dismissed with the back button.
+///
+/// ```dart
+/// import 'package:flutter/material.dart';
````
I think we should move this code sample to its own file in the examples repository and reference it here instead.

victorsanni replied:
i.e add a new file `examples/api/lib/widgets/routes/local_history_entry.0.dart` and a corresponding test file  `examples/api/test/widgets/routes/local_history_entry.0_test.dart`

AbdeMohlbi replied:
done

AbdeMohlbi committed:
Merge branch 'flutter:master' into abdemohlbi_issue_11660

AbdeMohlbi committed:
Add LocalHistoryEntry example for Flutter

This sample demonstrates the use of LocalHistoryEntry to manage temporary route-like state in a Flutter application. It includes a button to open a custom bottom panel that can be closed using the back button.

AbdeMohlbi committed:
Update local_history_entry.0.dart

AbdeMohlbi committed:
Add widget test for PanelDemo functionality

AbdeMohlbi committed:
Update local_history_entry.0.dart

AbdeMohlbi committed:
Update documentation for LocalHistoryEntry example

Updated documentation to replace 'dartpad' with 'sample' and added reference to example code location.

AbdeMohlbi committed:
update sample link to use /
